File: library_frontend/lib/Models-Providers/borrowrecord.dart
class BorrowRecord {
  final int id, bookId;
  final String title, author;
  final DateTime borrowedAt;
  final DateTime? returnedAt;

  BorrowRecord({
    required this.id,
    required this.bookId,
    required this.title,
    required this.author,
    required this.borrowedAt,
    this.returnedAt,
  });

factory BorrowRecord.fromJson(Map<String, dynamic> j) => BorrowRecord(
  id: (j['id']) ?? 0,         
  bookId: j['bookId'] ?? 0,  
  title: j['title'] as String? ?? '',
  author: j['author'] as String? ?? '',
  borrowedAt: DateTime.parse(j['borrowedAt'] as String),
  returnedAt: j['returnedAt'] == null
      ? null
      : DateTime.parse(j['returnedAt'] as String),
);
}

File: LibraryApp/Models/BorrowRecord.cs
namespace LibraryApp.Models
{
    public class BorrowRecord
    {
        public int Id { get; set; }
        public int BookId { get; set; }
        public Book Book { get; set; }

        public int UserId { get; set; }
        public User User { get; set; }

        public DateTime BorrowedAt { get; set; }
        public DateTime? ReturnedAt { get; set; }
    }

}
